show-expected-fail-tests.sh

```diff
@@ -0,0 +1,17 @@
+#! /bin/bash
+
+results_file=$1
+tests_to_add=$(grep ' # TODO passed but expected fail' ${results_file} | sed -E 's/^ok [0-9]+ (\(expected fail\) )?//' | sed -E 's/( \([0-9]+ subtests\))? # TODO passed but expected fail$//')
+
+fail_build=0
+while read -r test_id; do
```

Comment:
This is awesome! But the printing looks like:

```
ERROR: Passed test not present in testfile: POST /register rejects registration of usernames with '!'
ERROR: Passed test not present in testfile: POST /register rejects registration of usernames with '"'
ERROR: Passed test not present in testfile: POST /register rejects registration of usernames with ':'
ERROR: Passed test not present in testfile: POST /register rejects registration of usernames with '?'
ERROR: Passed test not present in testfile: POST /register rejects registration of usernames with '\'
ERROR: Passed test not present in testfile: POST /register rejects registration of usernames with '@'
ERROR: Passed test not present in testfile: POST /register rejects registration of usernames with '['
...
```

which makes it hard to copy the new names into the test file. I would say that if you find one, you should print out something along the lines of "ERROR: The following passed tests are not present in the testfile. Please add them:" and then print them out line by line individually.

Reply:
Done. They now look like:

```
#!/bin/bash -eo pipefail
./show-expected-fail-tests.sh /logs/results.tap
ERROR: The following passed tests are not present in testfile. Please append them to the file:
local user can join room with version 5
User can invite local user to room with version 5

WARN: Tests in testfile still marked as expected fail:
POST /register rejects registration of usernames with '£'
POST /register rejects registration of usernames with 'é'
Should reject keys claiming to belong to a different user
Enabling an unknown default rule fails with 404

Exited with code 1
```

Reply:
Looks great!

Reply:
I'm looking into the bug behind that second output now.